### PR TITLE
Fixing missing support for "Subcategory Levels" parameter in contacts category view

### PR DESCRIPTION
**Pull request by artur-stepien**
This adds the support for `Subcategory Levels`  in `Category` tab of  `List Contacts in a Category` view. Currently, this setting is not applied in the query, so using it will not allow viewing all contacts in the category and its subcategories.

### Summary of Changes
Added state entry and query filter for this parameter.

### Testing Instructions
- Create a contact category called `All contacts`
- Create a contact category called `A` and set the parent as `All contacts`
- Create a contact category called `B` and set the parent as `All contacts`
- Create contact `Contact from A` and place it in `A` category
- Create contact `Contact from B` and place it in `B` category
- Create a menu item of type `List Contacts in a Category`, chose category `All contacts` and make sure in `Category` tab `Subcategory Levels` is set to `All` or any positive value
- Save and head to the recently created menu item in front-end

### Actual result BEFORE applying this Pull Request
There should be no contacts displayed in this view.

### Expected result AFTER applying this Pull Request
All contacts from category `A` and `B` will be displayed.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by Quy** [comment]
Please fix phpcs: https://ci.joomla.org/joomla/joomla-cms/69748/1/8

**Comment by artur-stepien** [comment]
> Please fix phpcs: https://ci.joomla.org/joomla/joomla-cms/69748/1/8

Yeah, but what is it complaining about. The lack of alignment between `=` signs?

**Comment by ceford** [comment]
I have tested this item :white_check_mark: successfully on 696b25d20b2e963c94f3307d432c6cd35fbf832b<br /><br />Without the patch I see just the Subcategories list. With the patch I see the actual contacts in Categories A and B.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41618">issues.joomla.org/tracker/joomla-cms/41618</a>.</sub>

**Comment by artur-stepien** [comment]
@Quy Do you know why the test is failing?

**Comment by HLeithner** [comment]
This pull request has been automatically rebased to 5.1-dev.

**Comment by artur-stepien** [comment]
@HLeithner Do you know why this issue fails build? Its a bug that affects one of clients websites and I see no reply to the question.

File: components/com_contact/src/Model/CategoryModel.php [hunk 1]
```diff
@@ -177,7 +177,38 @@ protected function getListQuery()
             ->whereIn($db->quoteName('a.access'), $groups);
 
         // Filter by category.
-        if ($categoryId = $this->getState('category.id')) {
+        $categoryId           = $this->getState('category.id');
+        $includeSubcategories = (int)$this->getState('filter.max_category_levels', 1) !== 0;
```
Comment (Quy) [inline components/com_contact/src/Model/CategoryModel.php:181]:
```suggestion
        $includeSubcategories = (int) $this->getState('filter.max_category_levels', 1) !== 0;
```

File: components/com_contact/src/Model/CategoryModel.php [hunk 2]
```diff
@@ -177,7 +177,38 @@ protected function getListQuery()
...
+
+        if ($includeSubcategories) {
+            $categoryId = (int) $categoryId;
+            $levels     = (int) $this->getState('filter.max_category_levels', 1);
```
Comment (Quy) [inline components/com_contact/src/Model/CategoryModel.php]:
```suggestion
        $categoryId           = (int) $this->getState('category.id');
        $includeSubcategories = (int) $this->getState('filter.max_category_levels', 1) !== 0;

        if ($includeSubcategories) {
            $levels = (int) $this->getState('filter.max_category_levels', 1);
```

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on 8ececc41c04a671f0b798fb786afc1998d1138b4<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41618">issues.joomla.org/tracker/joomla-cms/41618</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41618">issues.joomla.org/tracker/joomla-cms/41618</a>.</sub>

**Comment by Razzo1987** [comment]
Thanks

**Pull request merged**